src/python/WMCore/Services/WMArchive/CMSSWMetrics.py

```diff
@@ -0,0 +1,509 @@
+            "Timing-tstoragefile-construct-totalMsecs": 11419,
+            "Timing-root-writev-totalMsecs": 0,
+            "Timing-file-construct-maxMsecs": 0.000291,
+            "Timing-file-prefetch-minMsecs": 0,
+            "Timing-root-stagein-numOperations": 1,
+            "Timing-file-close-minMsecs": 0.190476,
+            "Timing-file-construct-numOperations": 2,
+            "Timing-file-construct-numSuccessfulOperations": 2,
+            "Timing-root-stagein-minMsecs": 4.10695,
+            "Timing-root-write-totalMegabytes": 0,
+            "Timing-root-read-numSuccessfulOperations": 57,
+            "Timing-tstoragefile-writeActual-minMsecs": 0.005877,
+            "Timing-tstoragefile-readViaCache-numOperations": 3,
+            "Timing-tstoragefile-readActual-maxMsecs": 4168.78,
+            "Timing-file-close-numOperations": 1,
+            "Timing-file-position-numOperations": 14759,
+            "Timing-tstoragefile-read-totalMsecs": 2075,
+            "Timing-root-position-maxMsecs": 0.001203,
+            "Timing-root-construct-minMsecs": 0.000142,
+            "Timing-tstoragefile-readViaCache-numSuccessfulOperations": 0,
+            "Timing-file-position-numSuccessfulOperations": 14759,
+            "Timing-tstoragefile-writeViaCache-maxMsecs": 0,
+            "Timing-tstoragefile-writeViaCache-totalMegabytes": 0,
+            "Timing-file-write-numOperations": 7307,
+            "Timing-tstoragefile-flush-totalMsecs": 1847.36,
+            "Timing-root-construct-numOperations": 3,
+            "Timing-tstoragefile-seek-totalMsecs": 18.3395,
+            "Timing-tstoragefile-stat-totalMegabytes": 0,
+            "Timing-root-prefetch-totalMsecs": 0,
+            "Timing-root-prefetch-maxMsecs": 0,
+            "Timing-tstoragefile-writeActual-totalMsecs": 222.926
+        }
+}
+
+def CMSSWMetrics():
+    """
+    Helper function to convert CMSSW_METRICS dict values to their data-types
+    :return: dictionary
+    """
+    rdict = {}
+    for key, mdict in CMSSW_METRICS.items():
+        if key == 'XrdSiteStatistics':
+            rdict[key] = XrdMetrics(mdict)
+            continue
+        ndict = {}
+        for mkey, mval in mdict.items():
+            for ktype in [int, float, str, bool]:
+                if isinstance(mval, ktype):
+                    ndict[mkey] = ktype
+        rdict[key] = ndict
+    return rdict
+
+def XrdMetrics(mdict):
+    """
+    Convert given XrdSiteStatistics dict values to their data-types
+    :param mdict: XrdSiteStatistics dict
+    :return: dictionary
+    """
+    rdict = {}
+    for key, values in mdict.items():
+        entry = values[0]
+        ndict = {}
+        for mkey, mval in entry.items():
+            for ktype in [int, float, str, bool]:
+                if isinstance(mval, ktype):
+                    ndict[mkey] = ktype
+        rdict[key] = [ndict]
+    return rdict
+
+# main test: dump CMSSW metrics data-type value dict on stdout
+#if __name__ == '__main__':
```

Comment:
Why did you comment this main call? If this  script  is going to be used as  standard utils script, I'd rather keep it fully functional with a default `__main__`  toplevel call. 


Which brings another question. Why not splitting this script in two parts:
* One part providing the metric structure + functions to dump the metrics and keeping it under WMCore/Services 
* One part providing a basic import of the service and a tolevel  `__main__` call. And this one  we move it under `Utils`